src/concrete/ml/sklearn/base.py

```diff
@@ -1975,11 +1975,9 @@ def scatter1d(x, v, indices):
                     r = p
 
             # Return only the topk indexes
-            topk_labels = []
-            for i in range((self.n_neighbors)):
-                topk_labels.append(labels[i])
+            topk_labels = fhe_array(labels[:k])
 
-            return fhe_array(topk_labels)
+            return topk_labels
```

Comment:
Slicing is allowed in FHE.

So I just replaced the loop by fhe_array(labels[:k])